src/wallet/coinselection.cpp

```diff
@@ -115,36 +114,35 @@ bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& selectio
 
         // Backtracking, moving backwards
         if (backtrack) {
-            // Walk backwards to find the last included UTXO that still needs to have its omission branch traversed.
-            while (!curr_selection.empty() && !curr_selection.back()) {
-                curr_selection.pop_back();
-                curr_available_value += utxo_pool.at(curr_selection.size()).GetSelectionAmount();
+            if (curr_selection.empty()) { // No branch is untraversed. All solutions searched
+                break;
             }
 
-            if (curr_selection.empty()) { // We have walked back to the first utxo and no branch is untraversed. All solutions searched
-                break;
+            // Walk backwards to find the last included UTXO that still needs to have its omission branch traversed.
```

Comment:
Optional: Please update comment to "add omitted UTXOs back to lookahead before traversing the omission branch of last included UTXO"

Reply:
Done.